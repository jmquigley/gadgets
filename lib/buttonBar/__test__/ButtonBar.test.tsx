'use strict';

import * as assert from 'assert';
import {shallow} from 'enzyme';
import * as React from 'react';
import {ButtonBar, getDefaultButtonBarProps} from '../index';

test('Test retrieval of ButtonBar props object', () => {
	const props = getDefaultButtonBarProps();

	assert(props);
	expect(props).toMatchSnapshot();
});

test('Test creation of a simple ButtonBar instance', () => {
	const ctl = shallow(<ButtonBar className="test-class" />);

	assert(ctl)	;
	expect(ctl).toMatchSnapshot();
});

// TODO: add a test for each of the ButtonBar justifications
// TODO: add disable/visible test
