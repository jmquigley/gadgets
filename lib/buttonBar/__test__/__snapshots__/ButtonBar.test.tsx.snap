// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of a ButtonBar with center justify 1`] = `
<div
  className="ui-button-bar buttonBar normal"
>
  <div
    className="ui-button-bar-group buttonBarGroup center"
  />
</div>
`;

exports[`Test creation of a ButtonBar with left justify 1`] = `
<div
  className="ui-button-bar buttonBar normal"
>
  <div
    className="ui-button-bar-group buttonBarGroup left"
  />
</div>
`;

exports[`Test creation of a ButtonBar with right justify 1`] = `
<div
  className="ui-button-bar buttonBar normal"
>
  <div
    className="ui-button-bar-group buttonBarGroup right"
  />
</div>
`;

exports[`Test creation of a simple ButtonBar instance 1`] = `
<ButtonBar
  buttonSize="32px"
  className="test-class"
  contentEditable={false}
  direction="right"
  disabled={false}
  focus={false}
  height=""
  id=""
  justify={0}
  location="none"
  noedit={false}
  nohover={false}
  noripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  visible={true}
  width=""
>
  <div
    className="ui-button-bar buttonBar normal test-class"
  >
    <div
      className="ui-button-bar-group buttonBarGroup left"
    >
      <div
        className="buttonBarBox"
        key="0"
        style={
          Object {
            "height": "24px",
            "width": "24px",
          }
        }
      >
        <Button
          className=""
          contentEditable={false}
          direction="right"
          disabled={false}
          focus={false}
          height=""
          iconName="bomb"
          iconStyle=""
          id=""
          location="none"
          noedit={false}
          nohover={false}
          noripple={false}
          onClick={[Function]}
          selected={false}
          sizing="normal"
          style={Object {}}
          testing={false}
          visible={true}
          width=""
        >
          <div
            className="ui-button button ripple normal"
            onClick={[Function]}
            style={Object {}}
          >
            <Icon
              className="icon"
              contentEditable={false}
              direction="right"
              disabled={false}
              focus={false}
              height=""
              iconName="bomb"
              id=""
              imageFile=""
              location="none"
              noedit={false}
              nohover={false}
              noripple={false}
              selected={false}
              sizing="normal"
              style={Object {}}
              testing={false}
              visible={true}
              width=""
            >
              <i
                className="ui-icon icon none fa fa-fw fa-bomb normal"
                style={Object {}}
              />
            </Icon>
          </div>
        </Button>
      </div>
      <div
        className="buttonBarBox"
        key="1"
        style={
          Object {
            "height": "24px",
            "width": "24px",
          }
        }
      >
        <Button
          className=""
          contentEditable={false}
          direction="right"
          disabled={false}
          focus={false}
          height=""
          iconName="bomb"
          iconStyle=""
          id=""
          location="none"
          noedit={false}
          nohover={false}
          noripple={false}
          onClick={[Function]}
          selected={false}
          sizing="normal"
          style={Object {}}
          testing={false}
          visible={true}
          width=""
        >
          <div
            className="ui-button button ripple normal"
            onClick={[Function]}
            style={Object {}}
          >
            <Icon
              className="icon"
              contentEditable={false}
              direction="right"
              disabled={false}
              focus={false}
              height=""
              iconName="bomb"
              id=""
              imageFile=""
              location="none"
              noedit={false}
              nohover={false}
              noripple={false}
              selected={false}
              sizing="normal"
              style={Object {}}
              testing={false}
              visible={true}
              width=""
            >
              <i
                className="ui-icon icon none fa fa-fw fa-bomb normal"
                style={Object {}}
              />
            </Icon>
          </div>
        </Button>
      </div>
    </div>
  </div>
</ButtonBar>
`;

exports[`Test disabling of the ButtonBar control 1`] = `
<ButtonBar
  className=""
  contentEditable={false}
  direction="right"
  disabled={true}
  focus={false}
  height=""
  id=""
  justify={0}
  location="none"
  noedit={false}
  nohover={false}
  noripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  visible={true}
  width=""
>
  <div
    className="ui-button-bar buttonBar normal disabled nohover"
  >
    <div
      className="ui-button-bar-group buttonBarGroup left"
    >
      <div
        className="buttonBarBox"
        key="0"
        style={
          Object {
            "height": "24px",
            "width": "24px",
          }
        }
      >
        <Button
          className=""
          contentEditable={false}
          direction="right"
          disabled={true}
          focus={false}
          height=""
          iconName="bomb"
          iconStyle=""
          id=""
          location="none"
          noedit={false}
          nohover={false}
          noripple={false}
          onClick={[Function]}
          selected={false}
          sizing="normal"
          style={Object {}}
          testing={false}
          visible={true}
          width=""
        >
          <div
            className="ui-button button normal disabled nohover"
            onClick={[Function]}
            style={Object {}}
          >
            <Icon
              className="icon"
              contentEditable={false}
              direction="right"
              disabled={false}
              focus={false}
              height=""
              iconName="bomb"
              id=""
              imageFile=""
              location="none"
              noedit={false}
              nohover={false}
              noripple={false}
              selected={false}
              sizing="normal"
              style={Object {}}
              testing={false}
              visible={true}
              width=""
            >
              <i
                className="ui-icon icon none fa fa-fw fa-bomb normal"
                style={Object {}}
              />
            </Icon>
          </div>
        </Button>
      </div>
      <div
        className="buttonBarBox"
        key="1"
        style={
          Object {
            "height": "24px",
            "width": "24px",
          }
        }
      >
        <Button
          className=""
          contentEditable={false}
          direction="right"
          disabled={true}
          focus={false}
          height=""
          iconName="bomb"
          iconStyle=""
          id=""
          location="none"
          noedit={false}
          nohover={false}
          noripple={false}
          onClick={[Function]}
          selected={false}
          sizing="normal"
          style={Object {}}
          testing={false}
          visible={true}
          width=""
        >
          <div
            className="ui-button button normal disabled nohover"
            onClick={[Function]}
            style={Object {}}
          >
            <Icon
              className="icon"
              contentEditable={false}
              direction="right"
              disabled={false}
              focus={false}
              height=""
              iconName="bomb"
              id=""
              imageFile=""
              location="none"
              noedit={false}
              nohover={false}
              noripple={false}
              selected={false}
              sizing="normal"
              style={Object {}}
              testing={false}
              visible={true}
              width=""
            >
              <i
                className="ui-icon icon none fa fa-fw fa-bomb normal"
                style={Object {}}
              />
            </Icon>
          </div>
        </Button>
      </div>
    </div>
  </div>
</ButtonBar>
`;

exports[`Test making the ButtonBar invisible 1`] = `
<ButtonBar
  className=""
  contentEditable={false}
  direction="right"
  disabled={false}
  focus={false}
  height=""
  id=""
  justify={0}
  location="none"
  noedit={false}
  nohover={false}
  noripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  visible={false}
  width=""
>
  <div
    className="ui-button-bar buttonBar normal invisible"
  >
    <div
      className="ui-button-bar-group buttonBarGroup left"
    />
  </div>
</ButtonBar>
`;

exports[`Test retrieval of ButtonBar props object 1`] = `
Object {
  "className": "",
  "contentEditable": false,
  "direction": "right",
  "disabled": false,
  "focus": false,
  "height": "",
  "id": "",
  "justify": 0,
  "location": "none",
  "noedit": false,
  "nohover": false,
  "noripple": false,
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": false,
  "visible": true,
  "width": "",
}
`;
